src/libraries/Common/src/Interop/Unix/System.Native/Interop.GNameUName.cs

```diff
@@ -0,0 +1,73 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+using System.Buffers;
+using System.Text;
+using System;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        /// <summary>
+        /// Gets the group name associated to the specified group ID.
+        /// </summary>
+        /// <param name="gid">The group ID.</param>
+        /// <returns>On success, return a string with the group name. On failure, returns a null string.</returns>
+        internal static string? GetGName(uint gid) => GetGNameOrUName(gid, isGName: true);
+
+        /// <summary>
+        /// Gets the user name associated to the specified user ID.
+        /// </summary>
+        /// <param name="uid">The user ID.</param>
+        /// <returns>On success, return a string with the user name. On failure, returns a null string.</returns>
+        internal static string? GetUName(uint uid) => GetGNameOrUName(uid, isGName: false);
+
+        private static string? GetGNameOrUName(uint id, bool isGName)
+        {
+            // Common max name length, like /etc/passwd, useradd, groupadd
+            int outputBufferSize = 32;
+
+            while (true)
+            {
+                byte[] buffer = ArrayPool<byte>.Shared.Rent(outputBufferSize);
+                try
+                {
+                    int resultLength = isGName ?
+                        Interop.Sys.GetGName(id, buffer, buffer.Length) :
+                        Interop.Sys.GetUName(id, buffer, buffer.Length);
+
+                    if (resultLength < 0)
+                    {
+                        // error
+                        return null;
+                    }
+                    else if (resultLength < buffer.Length)
```

Comment:
Should this be `<=`? If it completely filled the buffer, shouldn't we be able to use it?